### DISCUSSION TITLE
How to get the gradient with respect to trajectory (Omega)?

### DISCUSSION THREAD

**Discussion by hkaggarwal** [mmuckley/torchkbnufft#62] [Q&A] [answered]
Hello @mmuckley,

Thank you for creating this NUFFT toolbox.  

I am trying to get the gradient with respect to the trajectory Omega. 
I have tried setting requires_grad=True for the trajectory. I have also tried making the trajectory as a part of model parameters by wrapping into nn.Parameter(ktraj) .  But somehow, I am not able to get the gradient. 

Can you please tell me, if it is possible to get the gradient with respect to trajectory in the torchkbnufft toolbox. 

Thank you,
Hemant

**Comment by mmuckley** [comment, marked as answer]
Hello @hkaggarwal, thanks for the compliment!

I don't think this is possible for the package. I think @guanhuaw might have spent some effort on this with his Bjork project. You might want to check those out.

Code: https://github.com/guanhuaw/Bjork
Paper: https://arxiv.org/abs/2101.11369

**Reply by hkaggarwal** [reply]
Thank you, @mmuckley, this is helpful.
